src/components/TokenClaimDialog.tsx

```diff
@@ -49,6 +56,35 @@ export default function TokenClaimDialog({
   const { t } = useTranslation()
   const { chainId } = useActiveWeb3React()
   const basicPools = useContext(BasicPoolsContext)
+  const userState = useContext(UserStateContext)
+  const gaugeData = useContext(GaugeContext)
+  const gaugesWithPoolName = useMemo<
+    ({
+      poolName: string
+      address: string
+      rewards: GaugeReward[]
+    } | null)[]
+  >(() => {
+    return Object.values(basicPools || {})
+      .map((pool) => {
+        const gauge = gaugeData.gauges[pool?.poolAddress || ""]
```

Comment:
poolAddress should always be defined here

Reply:
good catch ⚾